script/test/cmd/cmd_test.go

```diff
@@ -0,0 +1,775 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/joho/godotenv"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func fileNotExist(path string) bool {
+	return findExecutable(path) != nil
+}
+func findExecutable(file string) error {
+	d, err := os.Stat(file)
+	if err != nil {
+		return err
+	}
+	if m := d.Mode(); !m.IsDir() {
+		return nil
+	}
+	return os.ErrPermission
+}
+
+func binary(cmd string) string {
+	path, err := exec.LookPath(cmd)
+	if err != nil || fileNotExist(cmd) {
+		home := os.Getenv("HOME")
+		if home == "" {
+			fmt.Println("No $HOME environment variable found")
+		}
+	}
+	return path
+}
+
+//Git get the branch
+func Git() (branch string, error error) {
+	cmdName := "git"
+	cmdArgs := []string{"rev-parse", "--abbrev-ref", "HEAD"}
+	cmdOut, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Println("")
+		return "", errors.Wrap(err, "Error getting branch name")
+	}
+	branch = string(cmdOut)
+	return branch, nil
+}
+
+//UpdateUri Replaces variables with current branch and uri
+func UpdateURI(uri, branch, inputfile, output string) error {
```

Comment:
So, currently this function is tied to replacing only uri and branch, what do you think about making this more generic?
So, as an input we accept a map[string]string, in which pass key value pairs that we want to replace, which in this case will be, `"%URI%": uri`, and `"%REF", branch`, and then we iterate over the map and replace all the occurrences; this way we can replace any parameter from now.